.github/workflows/generate.yml

```diff
@@ -0,0 +1,57 @@
+name: Generate
+
+on:
+  schedule:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
+    # Runs every Monday at 00:00
+    - cron:  '0 0 * * 1'
+
+jobs:
+  generate:
+    permissions:
+        contents: write
+        pull-requests: write
+    runs-on: ubuntu-latest
+    env:
+      # the version of deno used by Run on Slack (v1.31.1)
+      deno-version: v1.31.1
+
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+
+      - name: Setup Deno ${{ env.deno-version }}
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: ${{ env.deno-version }}
+      
+      - name: Setup Git config
+        run: |
+            git config user.name github-actions
+            git config user.email github-actions@github.com
+
+      - name: Set BRANCH_NAME
+        # example: BRANCH_NAME=generate_2023-07-18_18-09
+        run: echo "BRANCH_NAME=generate_$(date +"%Y-%m-%d_%H-%M")" >> $GITHUB_ENV
+
+      - name: Run generation script
+        run: ./scripts/generate
+      
+      - name: Commit changes
+        id: commit
+        continue-on-error: true
+        run: |
+            git checkout -b $BRANCH_NAME
+            git commit -a -m "Automated commit: Latest generated changes from schedule action"
+
+      - name: Create Pull Request
+        if: ${{ steps.commit.outcome == 'success' }}
+        run: |
+            git push --set-upstream origin $BRANCH_NAME
+            gh pr create -r "slackapi/denosaurs" --label "enhancement" --title "Latest generated changes" --body \
+            "## 🚀 Automated Pull Request: Coded-Connectors Update
+            This PR brings the latest changes to coded-connectors, fueled by Slack's latest data. Developers can now access the newest features effortlessly.
+
+            *Github-Actions* 🤖"
+        env:
+           GH_TOKEN: ${{ github.token }}
```

Comment:
Whose token is this?

Reply:
From what I understand each repo is assigned a `GitHub app` that is used in the runners, it is its token

`github.token` description ([source](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context))
> A token to authenticate on behalf of the GitHub App installed on your repository. This is functionally equivalent to the GITHUB_TOKEN secret. For more information, see "[Automatic token authentication](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)."
Note: This context property is set by the Actions runner, and is only available within the execution steps of a job. Otherwise, the value of this property will be null.

In the repo settings you can provide it with `write` permissions for PRs

Reply:
Cool!